sasview: init at 4.1.2

rprospero opened the pull request:
###### Motivation for this change

[Sasview](http://www.sasview.org) is a data fitting and analysis package for small angle scattering data.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

rprospero committed:
pythonPackages.periodictable: init at 1.5.0

This package is a required dependency of SasView, which will be added
in a later commit.

rprospero committed:
pythonPackages.unittest-xml-reporting: init at 2.1.1

This package is used by Sasview, which will be added in a later commit.

rprospero committed:
pythonPackage.bumps: init at 0.7.6

This package is required by SasView, which will be added in a later commit.

rprospero committed:
pythonPackages.sasmodels: init at 0.96

Sasmodels provides a library of small angle scattering calculations
that are needed by Sasview, which will be added in my next commit.

rprospero committed:
sasview: init at 4.1.2

Sasview is a data fitting and analysis package for small angle
scattering data.  More information can be found at
https://www.sasview.org.

There is some unfortunate cruft to this commit.  SasView uses the
xhtml2pdf python package, which has a specific version of html5lib as
a dependency.  This module manually loads that version into sasview.
I haven't made xhtml2pdf available as its own package due to these
circumstances.

FRidh reviewed (changes requested):
Libraries that are in `python-packages.nix` should be used. If those versions are not compatible, then they can be overridden.

rprospero committed:
Remove explicit licenses import from python package sasmodels

The licenses variable was being required by sasmoels, but licenses was
already a child of the lib parameter and was redundant.

rprospero committed:
Make xhtml2pdf its own package in python-packages

xhtml2pdf now uses the main package html5lib with just an override to
pull the correct version number.

rprospero commented:
I'm now using html5lib from python-packages and merely overriding the version.  This has also allowed me to move xhtml2pdf into the python-packages.nix, instead of making it part of sasview.

I've also done just a bit of cleanup as I was manually importing licenses into a couple of the packages for no good reason.

rprospero committed:
Fix typo in license for xhtml2pdf python package

FRidh commented on pkgs/applications/science/misc/sasview/default.nix:
```diff
@@ -0,0 +1,45 @@
+{lib, fetchgit, gcc, python}:
+
+python.pkgs.buildPythonApplication rec {
+  name = "sasview-${version}";
```
The `name` attribute is added by `buildPython*` and should therefore be removed. Instead use `pname` for just the name.

FRidh commented on pkgs/applications/science/misc/sasview/default.nix:
```diff
@@ -0,0 +1,45 @@
+  };
+
+  patches = [./pyparsing-fix.patch ./local_config.patch];
+  sandbox = true;
```
sandbox?

FRidh commented on pkgs/development/python-modules/bumps/default.nix:
```diff
@@ -0,0 +1,25 @@
+{ stdenv, buildPythonPackage, fetchPypi, six}:
+
+buildPythonPackage rec {
+  name = "${pname}-${version}";
```
The `name` attribute is added by `buildPython*` and should therefore be removed.

FRidh commented on pkgs/development/python-modules/sasmodels/default.nix:
```diff
@@ -0,0 +1,23 @@
+{lib, fetchgit, buildPythonPackage, pytest, numpy, scipy, matplotlib, docutils}:
+
+buildPythonPackage rec {
+  name = "sasmodels-${version}";
```
The `name` attribute is added by `buildPython*` and should therefore be removed.

FRidh commented on pkgs/development/python-modules/sasmodels/default.nix:
```diff
@@ -0,0 +1,23 @@
+  name = "sasmodels-${version}";
+  version = "0.96";
+
+  propagatedBuildInputs = [docutils matplotlib numpy pytest scipy];
```
test dependencies should not be propagated

FRidh commented on pkgs/development/python-modules/xhtml2pdf/default.nix:
```diff
@@ -0,0 +1,33 @@
+
+let
+  #xhtml2pdf specifically requires version "1.0b10" of html5lib
+  html5 = html5lib.overrideAttrs( oldAttrs: rec{
```
We permit only a single version within `python-packages.nix`, so no overriding here.

FRidh replied:
Considering sasview is the only consumer of this package, I suggest moving the expression there.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -12276,6 +12278,26 @@ in {
     };
   };
 
+  periodictable = buildPythonPackage rec{
```
We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -12276,6 +12278,26 @@ in {
   };
 
+  periodictable = buildPythonPackage rec{
+    name = "${pname}-${version}";
```
The `name` attribute is added by `buildPython*` and should therefore be removed.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -17329,6 +17353,29 @@ in {
     };
   };
 
+  unittest-xml-reporting = buildPythonPackage rec {
```
We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -17329,6 +17353,29 @@ in {
   };
 
+  unittest-xml-reporting = buildPythonPackage rec {
+    name = "${pname}-${version}";
```
The `name` attribute is added by `buildPython*` and should therefore be removed.

rprospero committed:
Remove lint from Sasview package

dotlambda commented:
@matthewbauer This one created a conflict. I manually merged master into staging right now.
